File: app/src/main/java/com/example/rafael/myapplication/DetailContact.java
package com.example.rafael.myapplication;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.TextView;

import java.util.List;

public class DetailContact extends AppCompatActivity {

    Contact contact;

    private RecyclerView recyclerView;
    private PhoneAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail_contact);

        contact = (Contact) getIntent().getSerializableExtra("contact");

        recyclerView = findViewById(R.id.rvphone);

        mAdapter = new PhoneAdapter(contact.getListPhone());
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(mLayoutManager);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.addItemDecoration(new DividerItemDecoration(this, LinearLayoutManager.VERTICAL));
        recyclerView.setAdapter(mAdapter);

        mAdapter.notifyDataSetChanged();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        finish();
    }
}

File: app/src/main/java/com/example/rafael/myapplication/DetailContact.kt
package com.example.rafael.myapplication

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.rafael.myapplication.databinding.ActivityDetailContactBinding

class DetailContact : AppCompatActivity() {
    var contact: Contact? = null
    private var mAdapter: PhoneAdapter? = null
    lateinit var binding: ActivityDetailContactBinding

    companion object {
        const val EXTRA_CONTACT = "contact"

        fun launchIntent(context: Context, contact: Contact): Intent {
            val intent = Intent(context, DetailContact::class.java)
            intent.putExtra(EXTRA_CONTACT, contact)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDetailContactBinding.inflate(layoutInflater)
        setContentView(binding.root)

        contact = intent.getSerializableExtra(EXTRA_CONTACT) as? Contact
        bindViews()
    }

    private fun bindViews() = with(binding) {
        contact?.listPhones?.let {
            mAdapter = PhoneAdapter(it)

            recyclerView.apply {
                itemAnimator = DefaultItemAnimator()
                addItemDecoration(DividerItemDecoration(this@DetailContact, LinearLayoutManager.VERTICAL))
                adapter = mAdapter
            }
        }

        mAdapter?.notifyDataSetChanged()
    }

    override fun onDestroy() {
        super.onDestroy()
        finish()
    }
}